src/test/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfigSerializeTest.java

```diff
@@ -0,0 +1,247 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.config;
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig;
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.InjectableValues;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSet;
+import org.elasticsearch.common.settings.Settings;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+
+import static com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory.AUTHENTICATED;
+import static com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory.GRANTED_PRIVILEGES;
+
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+
+public class AuditConfigSerializeTest {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final WildcardMatcher DEFAULT_IGNORED_USER = WildcardMatcher.from(AuditConfig.DEFAULT_IGNORED_USERS);
+
+    @Before
+    public void setUp() {
+        InjectableValues.Std iv = new InjectableValues.Std();
+        iv.addValue(Settings.class, Settings.EMPTY);
+        objectMapper.setInjectableValues(iv);
+    }
+
+    @Test
+    public void testDefaultSerialize() throws JsonProcessingException {
+        // arrange
+        final AuditConfig audit = new AuditConfig(true, null, null);
+        // act
+        final String json = objectMapper.writeValueAsString(audit);
+        assertEquals("{" +
```

Comment:
I remember using the minidev JSONObject object to create JSON objects for testing in unit tests here - https://github.com/opendistro-for-elasticsearch/security/blob/master/src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/OpenDistroSecuritySSLReloadCertsActionTests.java#L67-L70. 

It helped make the tests little more readable for me. Just a suggestion in case your dealing with a lot of JSON object creation via testing

Reply:
@debjanibnrj Looks like minidev is some internal dependency. I used XContentBuilder and Jackson to assert all jsons and also removed the index field making it cleaner.